hax/hax/ha/thread.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import logging
+import threading as thr
+from typing import Dict, Optional
+
+from hax.ha.events import Event, EventListener
+from hax.ha.handler import EventHandler
+from hax.ha.handler.node import NodeEventHandler
+from hax.motr.planner import WorkPlanner
+from hax.types import StoppableThread
+from hax.util import ConsulUtil, InterruptedException, wait_for_event
+
+from ha.core.event_manager.subscribe_event import SubscribeEvent
+
+LOG = logging.getLogger('hax')
+
+
+class EventPollingThread(StoppableThread):
```

Comment:
Instead of polling, did we check if there's a callback mechanism with message bus where a callback function is invoked on whenever event is posted to the subscribed topic?

Reply:
[python-kafka](https://kafka-python.readthedocs.io/en/master/apidoc/KafkaConsumer.html) doesn't seem to provide this kind of interface. But probably @SwapnilGaonkar7 knows better as he is the author of EventListener